cpp/modules/luma.gl/core/src/glfw-animation-loop.cpp

```diff
@@ -40,31 +37,19 @@
 using namespace lumagl;
 using namespace lumagl::utils;
 
-GLFWAnimationLoop::GLFWAnimationLoop(const Size& size, const std::string& windowTitle, const wgpu::Device& device,
-                                     const wgpu::Queue& queue, const wgpu::BackendType backendType)
-    : AnimationLoop{size} {
-  this->_window = this->_initializeGLFW(backendType, windowTitle);
-
-  // NOTE: This **must** be done before any wgpu API calls as otherwise functions will be undefined
-  // TODO(ilija@unfolded.ai): Set this globally elsewhere
-  static bool procTableInitialized = false;
-  DawnProcTable procs = dawn_native::GetProcs();
-  if (!procTableInitialized) {
-    dawnProcSetProcs(&procs);
-    procTableInitialized = true;
-  }
+GLFWAnimationLoop::GLFWAnimationLoop(const Options& options) : AnimationLoop{options} {
+  this->_window = this->_initializeGLFW(options.backendType, options.windowTitle);
 
   // Create a device if none was provided
-  auto _device = device ? device : this->_createDevice(backendType);
-  auto _queue = queue ? queue : _device.CreateQueue();
+  auto _device = options.device ? options.device : this->_createDevice(options.backendType);
 
-  this->_binding = CreateBinding(backendType, this->_window, _device.Get());
+  this->_binding = glfw::CreateBinding(options.backendType, this->_window, _device.Get());
   if (!this->_binding) {
     throw new std::runtime_error("Failed to initialize GLFWAnimationLoop, no backends enable for luma.gl");
   }
 
   this->_swapchain = this->_createSwapchain(_device);
-  this->_initialize(_device, _queue);
+  this->_initialize(_device, options.queue);
```

Comment:
Nit: It is a bit disconcerting that we don't resolve the possibly null queue before we pass it down, but maybe changing that would drag up the contents of `_initialize` into this func.

Reply:
Wanted to consolidate that code in one place, but I agree that part should be handled more explicitly, updated